src/Task/SymfonyConsole.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GrumPHP\Task;
+
+use GrumPHP\Formatter\ProcessFormatterInterface;
+use GrumPHP\Runner\TaskResult;
+use GrumPHP\Runner\TaskResultInterface;
+use GrumPHP\Task\Config\ConfigOptionsResolver;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\GitPreCommitContext;
+use GrumPHP\Task\Context\RunContext;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Process\PhpExecutableFinder;
+
+/** @extends AbstractExternalTask<ProcessFormatterInterface> */
+class SymfonyConsole extends AbstractExternalTask
+{
+    public static function getConfigurableOptions(): ConfigOptionsResolver
+    {
+        return ConfigOptionsResolver::fromOptionsResolver(
+            (new OptionsResolver())
+                ->setDefaults([
+                    'bin' => './bin/console',
+                    'command' => [],
+                    'ignore_patterns' => [],
+                    'whitelist_patterns' => [],
+                    'triggered_by' => ['php', 'yml', 'xml'],
+                    'run_always' => false,
+                ])
+                ->addAllowedTypes('bin', ['string'])
+                ->addAllowedTypes('command', ['string[]'])
+                ->addAllowedTypes('ignore_patterns', ['array'])
+                ->addAllowedTypes('whitelist_patterns', ['array'])
+                ->addAllowedTypes('triggered_by', ['array'])
+                ->addAllowedTypes('run_always', ['bool'])
+                ->setAllowedValues(
+                    'bin',
+                    static fn(string $bin): bool => '' !== $bin
+                )
+                ->setAllowedValues(
+                    'command',
+                    static fn(array $command): bool => '' !== \implode('', $command)
```

Comment:
why do you implode here?
You could as well just check the array directly

Reply:
The implode would catch (some very bad config oversight) of `[""]` a non empty array with an empty string